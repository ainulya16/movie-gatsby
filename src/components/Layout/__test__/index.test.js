import React from 'react';
import { shallow } from 'enzyme';
import { StaticQuery } from 'gatsby';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import Component from '..';

const query = {
  site: {
    siteMetadata: {
      title: 'AIA Singapore',
      description: 'Frontend Web Skeleton',
      author: '@aia',
      keywords: ['aia', 'insurance'],
      menuLinks: [
        { name: 'our product', link: '/' },
        { name: 'life matters', link: '/' },
        { name: 'about aia', link: '/' },
        { name: 'help & support', link: '/' },
        { name: 'my aia', link: '/' },
      ],
    },
  },
  // static query for Logo on Navbar
  images: {
    edges: [
      {
        node: {
          relativePath: 'aia-logo-white.png',
          name: 'aia-logo-white',
          childImageSharp: {
            sizes: {
              base64:
                'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsSAAALEgHS3X78AAACZ0lEQVQ4y32UW4iNURTHv3MxTMYllzInjTGKya14EEbRpLyQ2+BpxqQhJbyNNDIuk9uLURIvrg8oClFEFPJizCA8kLxRSinyIJ3jt/RfxzpfsuvXXmvvtda391prf0mpVEps2Ay5RAN5LVyEPngPb+ExnIB5wS7n/sVisbxoSlZyMzyT825YAjNhNrTAMXgF16CQDuoBPdh2eA1Lk/8MCwBd8A7mxKAx+gZO+YC5WnoG8nLOeDBshoQ0zYeX0Ki9rG80wl0YLD1v+fBAzBZkql9Ra267CG6kr2B5afZgKpDvFWAs1PCRA8ydME57fptdsN4d6qE3BM8HeRZMsVm6FcjGJ6gNdgU+1uvKclihkzVobSiswmg18wVoDXl9qKD3YUIIugfqTGiFzTAadsJlAlm1J0MVnPITynFb6e/4YdfVusWZa0K35VCL1grn/3ypMsc9dmLJdfC9VDmsJ59Y1c1gC7SZA9zkdFag+lRL5XXVaWqPNXAE/Vwq8CTvJWuZr1q8BDOUr5xaah98sfZIPbse+TzV66qxxVpot76KlQtOTdpvgOGW17C3UQcaBR3RaR2MD/om5aRJ+giuOEzyVnhk7zvYL4QFMaAleqXk4/AZBuC2ctYXqvkN/Z4qvExr7fHZ+Y+hGu5AUUWwFrquvefWZ5I/6gAtyl9nOFjGH/wYlDcy/gCH4QUclOEANnslW/LPol/VP/InnEz/vizZO2AkTIdb9mmc2rTfD/v18SsWhPkoVKloHeWA5aiVlR0Ev2CxXxnnQ5LPWMH+4VNxwiT+9/QMT8NE6VaELsn2CLrD286GOiS/Aa/iv8rymE3HAAAAAElFTkSuQmCC',
              aspectRatio: 1,
              src:
                '/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png',
              srcSet:
                '/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w',
              srcWebp:
                '/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp',
              srcSetWebp:
                '/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp 80w',
              sizes: '(max-width: 80px) 100vw, 80px',
            },
          },
        },
      },
    ],
  },
};
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      ...query,
    }),
  );
});

describe('Layout Component', () => {
  it('should be exist', () => {
    const component = shallow(
      <Component data={query} title="Test Title">
        container
      </Component>,
    );
    expect(component.exists()).toBe(true);
  });

  it('should be use description', () => {
    const component = shallow(
      <Component data={query} description="description test">
        container
      </Component>,
    );
    expect(component.exists()).toBe(true);
  });

  it('should be match snapshot', () => {
    const tree = renderer
      .create(
        <Component data={query} title="Test Title">
          container
        </Component>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be match snapshot without navbar', () => {
    const tree = renderer
      .create(
        <Component data={query} showNavbar={false} title="Test Title">
          container
        </Component>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
