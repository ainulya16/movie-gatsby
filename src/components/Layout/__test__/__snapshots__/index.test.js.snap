// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should be match snapshot 1`] = `
Array [
  .c5 {
  color: #D31145;
  font-size: 2em;
}

.c4 {
  background: none;
  border: 1px solid #D31145;
  padding: 10px;
  line-height: 1;
}

.c4:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
  -webkit-transition: scale 0.1s ease-in-out;
  transition: scale 0.1s ease-in-out;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 0.4;
  -webkit-filter: alpha(opacity=40);
  filter: alpha(opacity=40);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #fff;
  box-shadow: 0 2px 4px 0 #eeeeee;
}

.c3 {
  border: none;
}

.c2 {
  height: 40px;
  margin: 18px;
}

.c1 {
  margin: 23px 0px 23px 40px;
}

.c1 button {
  padding: 5px;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.7);
}

.c1 button:hover {
  box-shadow: 0px 5px 15px 0px rgba(0,0,0,0.7);
}

.c1 i {
  font-size: 1.5em;
}

@media screen and (max-width:640px) {
  .c1 {
    margin-left: 20px;
    margin-right: 0px;
  }
}

<div
    style={
      Object {
        "minHeight": "calc(100vh - 70px)",
        "overflow": "hidden",
      }
    }
  >
    <nav
      className="c0"
      style={Object {}}
    >
      <div
        className="c1"
        onClick={[Function]}
      />
      <img
        className="c2"
        src="test-file-stub"
      />
      <button
        className="c3 c4"
        disabled={false}
        onClick={null}
        radius={null}
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <i
          className="icon-bubble-chat  c5"
          color="#D31145"
          size="m"
        />
      </button>
    </nav>
    <div
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      container
    </div>
  </div>,
  .c0:hover {
  color: #b3b1a8;
}

<div
    style={
      Object {
        "height": "8%",
      }
    }
  >
    <div
      className="footer"
    >
      <div />
      <div>
        <div
          className="container bg-b9 padding-0"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="text-center margin-top-s margin-top-sm-s margin-bottom-s margin-bottom-sm-s"
          >
            <a
              className="b5 bt5 c0"
              href="https://www.aia.com.sg/en/index/terms-of-use.html"
              target="_self"
            >
              Terms of Use
            </a>
            <span
              className="b5 link-seperator"
            >
               | 
            </span>
            <a
              className="b5 bt5 c0"
              href="https://www.aia.com.sg/en/index/personal-data-policy.html"
              target="_self"
            >
              AIA Personal Data Policy (Singapore)
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Layout Component should be match snapshot without navbar 1`] = `
Array [
  <div
    style={
      Object {
        "minHeight": "calc(100vh - 70px)",
        "overflow": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      container
    </div>
  </div>,
  .c0:hover {
  color: #b3b1a8;
}

<div
    style={
      Object {
        "height": "8%",
      }
    }
  >
    <div
      className="footer"
    >
      <div />
      <div>
        <div
          className="container bg-b9 padding-0"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="text-center margin-top-s margin-top-sm-s margin-bottom-s margin-bottom-sm-s"
          >
            <a
              className="b5 bt5 c0"
              href="https://www.aia.com.sg/en/index/terms-of-use.html"
              target="_self"
            >
              Terms of Use
            </a>
            <span
              className="b5 link-seperator"
            >
               | 
            </span>
            <a
              className="b5 bt5 c0"
              href="https://www.aia.com.sg/en/index/personal-data-policy.html"
              target="_self"
            >
              AIA Personal Data Policy (Singapore)
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
